### PR TITLE
Fix for the #409.

### PR DESCRIPTION
**Pull request by vladonemo**

### CODE REVIEW

File: Source/FluentMockContext.cs [hunk 1]
```diff
@@ -90,6 +90,7 @@ public MockInvocation LastInvocation
 
 		public void Dispose()
 		{
+			invocations.Reverse();
```
Comment (stakx) [inline Source/FluentMockContext.cs]: Out of curiosity (I haven't yet looked at this in detail), can you briefly explain why this works?

File: UnitTests/MockDefaultValueProviderFixture.cs [hunk 2]
```diff
@@ -93,6 +93,15 @@ public void CreatedMockIsVerifiedWithOwner()
 			Assert.Throws<MockVerificationException>(() => mock.Verify());
 		}
 
+		[Fact]
+		public void DefaultValueIsNotChangedWhenPerformingInternalInvocation()
+		{
+			var mockBar = new Mock<IBar> { DefaultValue = DefaultValue.Empty };
```
Comment (stakx) [inline UnitTests/MockDefaultValueProviderFixture.cs]: Given that `DefaultValue` is set to `DefaultValue.Empty`, would this test probably be more at home in `EmptyDefaultValueProviderFixture.cs`?

**Comment by stakx** [comment]
I'm sorry but this cannot be merged as is. Let's try this with a fresh start. Please check out a new feature branch splitting of from the current tip of this repo's `develop` branch. Then add the changes from commit 4b02b918ebc54be69b80a5613739892080d6fb7e to that feature branch (e.g. by `cherry-pick`ing). Then submit a new PR from that feature branch to Moq's `develop` branch. Thank you!

File: UnitTests/MockDefaultValueProviderFixture.cs (hunk 2 above)
Comment (vladonemo) [inline UnitTests/MockDefaultValueProviderFixture.cs]: I think that the MockDefaultValueProvuderFixture is better place for this test, as the EmptyDefalutValueProviderFixture tests the EmptyDefaultValueProvider class - unrelated to the scenario.

File: Source/FluentMockContext.cs (hunk 1 above)
Comment (vladonemo) [inline Source/FluentMockContext.cs]: I explained this in the new PR.

**Comment by vladonemo** [comment]
I have submitted a new PR - #411. I hope it is correct now :) Sorry for the mess ...

**Pull request closed without merging**